awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsApplicationSignalsCustomizerProvider.java

```diff
@@ -133,6 +179,67 @@ private SdkTracerProviderBuilder customizeTracerProviderBuilder(
     return tracerProviderBuilder;
   }
 
+  private SdkMeterProviderBuilder customizeMeterProvider(
+      SdkMeterProviderBuilder sdkMeterProviderBuilder, ConfigProperties configProps) {
+
+    if (isApplicationSignalsRuntimeEnabled(configProps)) {
+      Set<String> registeredScopeNames = new HashSet<>(1);
+      String jmxRuntimeScopeName = "io.opentelemetry.jmx";
+      registeredScopeNames.add(jmxRuntimeScopeName);
+
+      configureMetricFilter(configProps, sdkMeterProviderBuilder, registeredScopeNames);
+
+      MetricExporter metricsExporter =
+          ApplicationSignalsExporterProvider.INSTANCE.createExporter(configProps);
+      MetricReader metricReader =
+          ScopeBasedPeriodicMetricReader.create(metricsExporter, registeredScopeNames)
+              .setInterval(getMetricExportInterval(configProps))
+              .build();
+      sdkMeterProviderBuilder.registerMetricReader(metricReader);
+
+      logger.info("AWS Application Signals runtime metric collection enabled");
+    }
+    return sdkMeterProviderBuilder;
+  }
+
+  private static void configureMetricFilter(
+      ConfigProperties configProps,
+      SdkMeterProviderBuilder sdkMeterProviderBuilder,
+      Set<String> registeredScopeNames) {
+    Set<String> exporterNames =
+        DefaultConfigProperties.getSet(configProps, "otel.metrics.exporter");
+    if (exporterNames.contains("none")) {
+      for (String scope : registeredScopeNames) {
+        sdkMeterProviderBuilder.registerView(
+            InstrumentSelector.builder().setMeterName(scope).build(),
+            View.builder().setAggregation(Aggregation.defaultAggregation()).build());
+
+        logger.log(Level.FINE, "Registered scope {0}", scope);
+      }
+      sdkMeterProviderBuilder.registerView(
+          InstrumentSelector.builder().setName("*").build(),
+          View.builder().setAggregation(Aggregation.drop()).build());
+    }
```

Comment:
could you help to add some comment on this method? I am not sure if I get the intention correctly?
Does it mean that - If customer set metric_exporter to none, we will only keep the configured runtime metrics and drop everything else? Otherwise, we keep everything?

Reply:
`View` is registered when `metric_exporter` to `none` so that only runtime metrics are collected through MeterProvider. In other cases, we respect the original metric collection rules.